lib/interpreters/compute_several_solutions.rb

```diff
@@ -18,6 +18,16 @@
 
 module Interpreters
   class SeveralSolutions
+    def self.duplicate_service_vrp(service_vrp, vrp_hash = nil)
+      service_vrp.map{ |key, value|
+        if key == :vrp
+          vrp_hash ||= JSON.parse(value.to_json, symbolize_names: true)
```

Comment:
Why don't we use `vrp.as_json` instead of `JSON.parse(vrp.to_json, symbolize_names: true)`? I think `as_json` already returns a new hash object with each call and we can make sure that `as_json` returns proper "symbolized" keys, no? What do you think? That way, we skip two intermediate parse operation in `hash->to_json->parse->hash` -- depending on the size of the VRP, this might take time and memory.

Reply:
ruby as_json returns explicitly string keys : https://apidock.com/rails/ActiveModel/Serializers/JSON/as_json 
I probably need to introduce an option for our use case

Reply:
I am not sure if to_json minds if the hash has string or symbol keys but I just realised, we implemented the wrong function. We should have implemented `to_hash` (which calls `as_json` at the moment but we can delete that) instead of `as_json` because that is our intention -- our object is a glorified hash and it can be represented as a hash -- i..e, to_hash (which is different then `to_h`).

If we rename the `as_json` to `to_hash` (and remove the other `to_hash`) and have the hash have `:symbol` keys then the native `Hash::as_json` function will do the necessary for handling the string business and we can use `to_hash` for dump/load logic skipping the json logic (which is useful for file dumps). What do you say?
 
https://justincypret.com/blog/activesupport-as-json-vs-to-json#adapt-objects-to-hash